streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientEndpoint.scala

```diff
@@ -80,10 +80,15 @@ object FlinkClientEndpoint {
   def shutdown(request: ShutDownRequest): ShutDownResponse = {
     request.executionMode match {
       case YARN_SESSION => YarnSessionClient.shutdown(request)
-      case KUBERNETES_NATIVE_SESSION => {
-        if (K8sFlinkConfig.isV2Enabled) KubernetesSessionClientV2.shutdown(request)
-        else KubernetesNativeSessionClient.shutdown(request)
-      }
+      case KUBERNETES_NATIVE_SESSION =>
+        K8sFlinkConfig.isV2Enabled match {
+          case true => KubernetesSessionClientV2.shutdown(request)
+          case _ => KubernetesNativeSessionClient.shutdown(request)
+        }
+      case KUBERNETES_NATIVE_APPLICATION =>
+        K8sFlinkConfig.isV2Enabled match {
```

Comment:
This is still an incorrect pattern matching branch line.
When KUBERNETES_NATIVE_APPLICATION = true and K8sFlinkConfig.isV2Enables = false, a match missing error will be thrown here.